src/instructlab/data/generate_data.py

```diff
@@ -34,15 +32,113 @@ def gen_data(
+    from instructlab.process.process import add_process
+
+    add_process(
+        process_mode=process_mode,
+        process_type=ILAB_PROCESS_TYPES.DATA_GENERATION,
+        target=create_server_and_generate,
+        extra_imports=[
+            ("instructlab.configuration", "_serve", "_serve_vllm", "_serve_llama_cpp")
+        ],
+        serve_cfg=serve_cfg,
+        model_name=model_path,
+        num_cpus=num_cpus,
+        num_instructions_to_generate=sdg_scale_factor,
+        taxonomy=taxonomy_path,
+        taxonomy_base=taxonomy_base,
+        output_dir=output_dir,
+        console_output=not quiet,
+        endpoint_url=endpoint_url,
+        api_key=api_key,
+        yaml_rules=yaml_rules,
+        chunk_word_count=chunk_word_count,
+        server_ctx_size=server_ctx_size,
+        tls_client_cert=http_client_params.get("tls_client_cert")
+        if http_client_params.get("tls_client_cert") != ""
+        else None,
+        tls_client_key=http_client_params.get("tls_client_key")
+        if http_client_params.get("tls_client_key") != ""
+        else None,
+        tls_client_passwd=http_client_params.get("tls_client_passwd")
+        if http_client_params.get("tls_client_passwd") != ""
+        else None,
+        tls_insecure=http_client_params.get("tls_insecure")
+        if http_client_params.get("tls_insecure") != ""
+        else None,
+        model_family=model_family,
+        pipeline=pipeline,
+        enable_serving_output=enable_serving_output,
+        batch_size=batch_size,
+        gpus=gpus,
+        checkpoint_dir=checkpoint_dir,
+        max_num_tokens=max_num_tokens,
+        system_prompt=system_prompt,
+        use_legacy_pretraining_format=use_legacy_pretraining_format,
+        # http_client_params=http_client_params,
+    )
 
-    # pylint: disable=ungrouped-imports
-    from instructlab.sdg.utils import GenerateException
 
+# actual generate_data function to be kicked off in its own process/eventually be REST API endpoint
+# pylint: disable=redefined-outer-name
+def create_server_and_generate(
+    serve_cfg,
+    model_name,
+    num_cpus,
+    num_instructions_to_generate,
+    taxonomy,
+    taxonomy_base,
+    output_dir,
+    console_output,
+    endpoint_url,
+    api_key,
+    yaml_rules,
+    chunk_word_count,
+    server_ctx_size,
+    tls_client_cert,
+    tls_client_key,
+    tls_client_passwd,
+    tls_insecure,
+    model_family,
+    pipeline,
+    enable_serving_output,
+    batch_size,
+    gpus,
+    checkpoint_dir,
+    max_num_tokens,
+    system_prompt,
+    use_legacy_pretraining_format,
+    log_file,
+):
     backend_instance = None
+    # we need to use the instructlab logger so that the libraries inherit the config we set up.
+    logger = logging.getLogger("instructlab")
+
+    # First Party
+    from instructlab.log import add_file_handler_to_logger
+
+    add_file_handler_to_logger(log_file=log_file, logger=logger)
+    logger.setLevel(logging.INFO)
```

Comment:
What if a user has set a different log level previously? Why are we overwriting the level to `INFO`?

Reply:
good point! I can plumb some way here to only set it to the level in the config.